Rewrote the 'balance' function in utils.py to fix issue #106

adbugger opened the pull request:
Fixes #106.

The balance function has now been rewritten to maintain relative order of elements, and return all the elements in a balanced data list. The test cases have also been rewritten to reflect this fact.

marco-c commented on autowebcompat/utils.py:
```diff
@@ -140,32 +140,40 @@ def next(self):
-    queue_1 = []
-    queue_0 = []
+    # Initialise last_label to not (label of first element)
+    last_label = it[0][1]
```
The function needs to support an iterator, `it` could be a list or an iterator.
You should add a test case that passes to balance an iterator, so you can make sure it's working in both cases.

adbugger commented:
I realised that the initialisation of `last_label` to inverse of the first label was not needed. Changed both the `balance` function and added a test case for a list iterator.

marco-c commented on tests/test_utils.py:
```diff
@@ -88,9 +89,32 @@ def test_balance_unbalanced_data():
+
+    with pytest.raises(StopIteration):
+        next(balanced_data)
+
```
Add the iterator test in the same function, so you don't have to define the data twice. You could also reuse the assertions if you do something like this:
```Python
for input_type in [list, iter]:
    balanced_data = utils.balance(input_type(unbalanced_tuples))
    ...
```

marco-c replied:
There should be even a pytest way to parametrize the function, if you want.

marco-c commented on autowebcompat/utils.py:
```diff
@@ -140,32 +140,40 @@ def next(self):
-                last_label = e[1]
-                yield e
+            queue[cur_label].append(e)
+            other_label = 1 if cur_label == 0 else 0
```
I think this like and the following three lines are unneeded

marco-c commented on tests/test_utils.py:
```diff
@@ -55,22 +55,26 @@ def test_write_labels():
-        ("data4", 0),
-        ("data5", 0),
-        ("data6", 1)]
+data = [
```
Nit: Call this `unbalanced_data` or `test_balance_data` as `data` is too generic.
